src/index.ts

```diff
@@ -1,22 +1,36 @@
 import { DataFile } from "./data-file";
-import { Database } from "./database";
+import { Database, FieldType, containsType } from "./database";
 import { IndexFile } from "./index-file";
 import { RangeResolver } from "./resolver";
 
 export async function init(
-  dataUrl: string | RangeResolver,
-  indexUrl: string | RangeResolver
+	dataUrl: string | RangeResolver,
+	indexUrl: string | RangeResolver
 ) {
-  return Database.forDataFileAndIndexFile(
-    typeof dataUrl === "string"
-      ? DataFile.forUrl(dataUrl)
-      : DataFile.forResolver(dataUrl),
-    typeof indexUrl === "string"
-      ? await IndexFile.forUrl(indexUrl)
-      : await IndexFile.forResolver(indexUrl)
-  );
+	return Database.forDataFileAndIndexFile(
+		typeof dataUrl === "string"
+			? DataFile.forUrl(dataUrl)
+			: DataFile.forResolver(dataUrl),
+		typeof indexUrl === "string"
+			? await IndexFile.forUrl(indexUrl)
+			: await IndexFile.forResolver(indexUrl)
+	);
+}
+
+interface GlobalMap {
+	Appendable: {
+		init: Function;
+		FieldType: typeof FieldType;
+		containsType: typeof containsType;
+	};
+}
+
+declare global {
+	var Appendable: GlobalMap["Appendable"];
```

Comment:
some type gymnastics

Reply:
Interesting, I'm surprised this isn't automatic.